OpenOversight/app/main/views.py

```diff
@@ -305,6 +305,47 @@ def list_officer(department_id, page=1, from_search=False):
         from_search=from_search)
 
 
+@main.route('/department/<int:department_id>/chart')
+def dept_org_chart(department_id):
+    department = Department.query.filter_by(id=department_id).first()
+    if not department:
+        abort(404)
+
+    officers = Officer.query.filter(Officer.department_id == department_id).all()
+    # map each officer to their most recent assignment
+    assign_dict = {}
+    assign_records = Assignment.query.all()
+    for record in assign_records:
+        if record.officer_id not in assign_dict:
+            assign_dict[record.officer_id] = record
+        else:
+            if record.star_date > assign_dict[record.officer_id].star_date:
+                assign_dict[record.officer_id] = record
+
+    officer_nodes = {}
+    rank_types = ["NOT SURE"]
+
+    for officer in officers:
+        rec = assign_dict[officer.id]
+        if len(rec.rank) <= 1:
+            rec.rank = "NOT SURE"
+        safe_rank = rec.rank.upper()
+        if safe_rank not in rank_types:
+            # keep NOT SURE at the end so it will display last on page
+            rank_types.insert(0, safe_rank)
+        if safe_rank not in officer_nodes.keys():
+            officer_nodes[safe_rank] = []
+        else:
+            officer_nodes[safe_rank].append({'id': officer.id,
```

Comment:
This shouldn't be in an `else` block or it won't add the first officer of a rank it finds.

Reply:
ah ha! thanks for catching this, fixed in last commit